blocks/api/serializer.js

```diff
@@ -15,15 +15,45 @@ import { Component, createElement, renderToString, cloneElement, Children } from
 import { applyFilters } from '../hooks';
 import { getBlockType, getUnknownTypeHandlerName } from './registration';
 
+/**
+ * Returns the block's name with common prefixes: 'core/' or 'core-' (in 'core-embed/') dropped
+ *
+ * @param {String}   blockName  The block name
+ * @return {string}             Friendly name
+ */
+function friendlyBlockName( blockName ) {
+	return blockName.replace( /\//, '-' ).replace( /^core-/, '' );
+}
+
 /**
  * Returns the block's default classname from its name
  *
  * @param {String}   blockName  The block name
  * @return {string}             The block's default class
  */
 export function getBlockDefaultClassname( blockName ) {
-	// Drop common prefixes: 'core/' or 'core-' (in 'core-embed/')
-	return 'wp-block-' + blockName.replace( /\//, '-' ).replace( /^core-/, '' );
+	return `wp-block-${ friendlyBlockName( blockName ) }`;
+}
+
+/**
+ * Returns a random base 36 string, 36^7 possible strings, maximum seven chars
+ * According to the birthday paradox repetition occur after sqrt(36^7) = 279936 random strings are generated
```

Comment:
If this is random, is it really the case that repetition could only occur after this number of strings has been generated, or is it rather there's a 1 in 279936 chance that a collision could occur? Meaning a post with two blocks could potentially share the same ID (though with a very low chance).

Reply:
Sorry, @aduth I missed your comment. I changed the comments to be more accurate. The number I wanted to refer is the expected number of ids we have to generate (but I used a wrong formula) to have the first collision. I added the formula to compute the probability of collision given k strings, and the correct number of expected ids we have to generate before having the first collision. Both formulas come from this page https://en.wikipedia.org/wiki/Birthday_attack. A post with two blocks can share the same id but the probability is very very low 1.27*10^-11.